PYME/Acquire/xyztc.py

```diff
@@ -92,7 +110,8 @@ def __init__(self, scope, dim_order='XYCZT', stack_settings=None, time_settings=
         if channel_settings is None:
             self.shape_c = 1
         else:
-            self.shape_c = channel_settings.get('num_channels', 1)
+            #self.shape_c = channel_settings.get('num_channels', 1)
+            self.shape_c = getattr(channel_settings, 'num_channels', 1)
```

Comment:
@Yujin-Bao  and @David-Baddeley  , `channel_settings` is passed as an argument here in the base class and not stored as an attribute, so I'm finding this change a bit odd.

Was there meant to be a `self.channel_settings = channel_settings` on a previous line in this `__init__`? Or might it make sense to revert this change and patch the subclass that motivated it?

Reply:
No, this makes sense (at least in the absense of any prior definition of what `channel_settings` should look like. It's a `getattr` on the `channel_settings` object, not on the class itself. What we do need is a base class for `channel_settings` to document the interface. The reason for changing from a dict like access for `'num_channels'` in `channel_settings` to an attribute access was so that it could be a computed property rather than a fixed value.

@barentine - for your multi-colour camera use case all we would need is to define a channel settings like this:

```python
class ChannelSettings(object):
    @property
    def num_channels(self):
        return scope.cam.num_channels

```

(or similar)

Alternatively, use a `named_tuple` for a static `num_channels`

Reply:
Ah, OK. Thanks for the explanation and pointer, @David-Baddeley !